contracts/orbital-core/src/contract.rs

```diff
@@ -1,119 +1,171 @@
+    StdResult, Uint64,
 };
 
 pub const CONTRACT_NAME: &str = "orbital-core";
 pub const CONTRACT_VERSION: &str = env!("CARGO_PKG_VERSION");
 
+pub type OrbitalResult = NeutronResult<Response<NeutronMsg>>;
+pub type QueryDeps<'a> = Deps<'a, NeutronQuery>;
+pub type ExecuteDeps<'a> = DepsMut<'a, NeutronQuery>;
+
 #[entry_point]
 pub fn instantiate(
-    deps: DepsMut,
+    deps: ExecuteDeps,
     _env: Env,
     _info: MessageInfo,
     msg: InstantiateMsg,
-) -> Result<Response, OwnershipError> {
+) -> OrbitalResult {
     set_contract_version(deps.storage, CONTRACT_NAME, CONTRACT_VERSION)?;
     initialize_owner(deps.storage, deps.api, Some(&msg.owner))?;
 
+    USER_NONCE.save(deps.storage, &Uint64::zero())?;
     Ok(Response::new())
 }
 
 #[entry_point]
 pub fn execute(
-    deps: DepsMut,
+    deps: ExecuteDeps,
     env: Env,
     info: MessageInfo,
     msg: ExecuteMsg,
-) -> Result<Response, ContractError> {
+) -> NeutronResult<Response<NeutronMsg>> {
     match msg {
+        // admin action to manage ownership of orbital-core
         ExecuteMsg::UpdateOwnership(action) => {
-            admin_update_ownership(deps, &env.block, &info.sender, action)
+            admin::transfer_admin(deps, &env.block, &info.sender, action)
         }
+        // admin action to enable new domain for user registration
         ExecuteMsg::RegisterNewDomain {
             domain,
             account_type,
-        } => admin_register_new_domain(deps, info, domain, account_type),
-        ExecuteMsg::RegisterUser {} => register_user(deps, env, info),
+        } => admin::register_new_domain(deps, info, domain, account_type),
+        // user action to create a new user account which enables registration to domains
+        ExecuteMsg::RegisterUser {} => user::register(deps, env, info),
+        // user action to register a new domain which creates their clearing account
+        ExecuteMsg::RegisterUserDomain { domain } => {
+            user::register_new_domain(deps, env, info, domain)
+        }
     }
 }
 
-fn register_user(deps: DepsMut, _env: Env, info: MessageInfo) -> Result<Response, ContractError> {
-    // user can only register once
-    if USER_CONFIGS.has(deps.storage, info.sender.clone()) {
-        return Err(ContractError::UserAlreadyRegistered {});
+#[entry_point]
+pub fn query(deps: QueryDeps, _env: Env, msg: QueryMsg) -> StdResult<Binary> {
+    match msg {
+        QueryMsg::OrbitalDomain { domain } => to_json_binary(&query_orbital_domain(deps, domain)?),
+        QueryMsg::UserConfig { addr } => to_json_binary(&query_user_config(deps, addr)?),
+        QueryMsg::Ownership {} => to_json_binary(&query_ownership(deps)?),
+        QueryMsg::ClearingAccountAddress { addr, domain } => {
+            to_json_binary(&query_clearing_account(deps, domain, addr)?)
+        }
+        QueryMsg::UserAddresses {} => to_json_binary(&query_user_addresses(deps)?),
```

Comment:
Should be with pagination or it will run out of gas after adding too many users.

Reply:
sorry, should've noted this was only for testing/development, will remove it now